lib/widgets/selection.dart

```diff
@@ -1671,6 +1672,40 @@ class _AddToPlaylistSelectionAction extends StatelessWidget {
   }
 }
 
+/// Action that removes a song from playlist.
+class _FavoriteSelectionAction extends StatelessWidget {
+  const _FavoriteSelectionAction({ Key? key }) : super(key: key);
+
+  void _handleTap(BuildContext context, ContentSelectionController controller) {
+    final contentTuple = ContentUtils.selectionPack(controller.data);
+    FavoritesControl.instance.setFavorite(
+      contentTuple: contentTuple,
+      value: contentTuple.any((el) => !el.isFavorite),
+    );
+    controller.close();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = getl10n(context);
+    final controller = ContentSelectionController._of(context);
+    final theme = ThemeControl.instance.theme;
+    return _ActionBuilder(
+      controller: controller,
+      shown: () => true,
+      builder: (context, child) => EmergeAnimation(
+        animation: controller.animation,
+        child: HeartButton(
+          tooltip: l10n.addToFavorites,
+          active: controller.data.any((el) => !el.data.isFavorite),
+          onPressed: () => _handleTap(context, controller),
+          inactiveColor: theme.colorScheme.onSurface,
```

Comment:
Suggestion: I would suggest making this the active color as well here, to align with the other actions. The standard white would look better in my opinion.